### ISSUE TITLE
Add accessibility label to touchbar buttons

### ISSUE THREAD

**Issue by christopherpross** [microsoft/vscode#82154]
Hey,
the buttons in VSCode are not labeled for VoiceOver, because this buttons use only icons. I understand, that icons are better as text labels. I searched in the code of electron, whether we can easily use a accessibilityLabel, that VoiceOver can read out. This has the advantage that the design of the Touch Bar-elements haven't to change, only a attribute have to add.
I found out, that electron don't offer a accessibilityLabel for Touch Bar-controls, so I opened a issue.
[here](https://github.com/electron/electron/pull/20454) is the issue fixed, very fast. :-)
So, electron offers now an AccessibilityLabel for Touchbar-controls.
Hopefully this feature can be merged in VSCode.
When this is not possible, in the next time, a work-around can be, to use text labels, if the accessibility mode is turned on.
What do you think?

best regards,
Christopher

**Comment by bpasero** [comment]
Sounds good. It will probably be another 3-6 months before we go with a newer Electron version (>6) so we need to wait for that.

**Comment by Det87** [comment]
What's the Roadmap on updating Electrons? I just received my first 1.45 nightly, but it's still not got even Electron 8 (stable since 2 months ago: https://www.electronjs.org/releases/stable?page=2), and judging by the VSC release history (https://github.com/microsoft/vscode/releases), Electron 9 could still even make it to stable before the final 1.45 is out (https://www.electronjs.org/docs/tutorial/electron-timelines).

**Comment by bpasero** [comment]
@christopherpross unfortunately I do not see support for an accessibility label in the touchbar classes we use in VSCode: `TouchBarSegmentedControl` has an array of `SegmentedControlSegment` which do not provide any accessibility label:

![image](https://user-images.githubusercontent.com/900690/91410272-f4c30e00-e846-11ea-9638-93ddcf7d48f6.png)

Maybe you could ask for support in Electron repo.

**Comment by christopherpross** [comment]
Hm, so here is the issue I had created for this:
https://github.com/electron/electron/issues/20443

and this is the pr which resolves my issue, there is also a small description how it is now resolved:
https://github.com/electron/electron/pull/20454

I hope it helps.

all the best,
Christopher

**Comment by bpasero** [comment]
@christopherpross I am aware of the issue and the change, but VSCode is not using `TouchBarButton` or `TouchBarLabel` for the touchbar support. We need this API on the level of `TouchBarSegmentedControl`.

**Comment by isidorn** [comment]
Touchbar is going away on latest macs. thus closing as out of scope
